argoproj-labs/argocd-operator
# chore: enable pod security admission

**What type of PR is this?**

/kind chore

**What does this PR do / why we need it**:

Updates the code to work with Pod Security Admission, which is enabled by default in k8s 1.24

**Have you updated the necessary documentation?**


**How to test changes / Special notes to the reviewer**:

- Install the operator on a k8s 1.24 system.
- Install an Argo CD instance
```yaml
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: argocd
spec:
  applicationSet: {}
  notifications:
    enabled: true
  ha:
    enabled: false
  grafana:
    enabled: true
  redis:
    disableTLSVerification: false
  dex:
    config: |
      connectors:
      - type: mockCallback
        id: mock
        name: Example
```
- Ensure all workloads are running
```bash
% kubectl get po
NAME                                                READY   STATUS    RESTARTS   AGE
argocd-application-controller-0                     1/1     Running   0          34s
argocd-applicationset-controller-6d99f5bc48-cpbdm   1/1     Running   0          34s
argocd-dex-server-6f4bd74657-jl7zk                  1/1     Running   0          34s
argocd-grafana-7588bd47b5-8xrqt                     1/1     Running   0          34s
argocd-notifications-controller-86ddb6ff44-kqxjk    1/1     Running   0          33s
argocd-redis-7c79b9c5b6-7wmcf                       1/1     Running   0          34s
argocd-repo-server-857cf5d777-j6vnd                 1/1     Running   0          34s
argocd-server-579b655b48-c49rc                      1/1     Running   0          34s
```
- Install the operator on a k8s 1.23 system
- Install an ArgoCD instance
- Ensure all workloads are running
